src/clients/SpokePoolClient/SpokePoolClient.ts

```diff
@@ -492,15 +488,15 @@ export abstract class SpokePoolClient extends BaseAbstractClient {
       // For each deposit, resolve its quoteTimestamp to a block number on the HubPool.
       // Don't bother filtering for uniqueness; the HubPoolClient handles this efficienctly.
       const quoteBlockNumbers = await this.getBlockNumbers(
-        depositEvents.map(({ args }) => Number(args["quoteTimestamp"]))
+        depositEvents.map((e) => (e as DepositWithBlock).quoteTimestamp)
       );
       for (const event of depositEvents) {
```

Comment:
Could we apply the type assertion on the loop to avoid having to do it multiple places within?

```suggestion
      for (const event of depositEvents as DepositWithBlock[]) {
```